File: E-Tour/src/main/java/com/example/demo/entities/Tour.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tour")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Tour {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tour_id")
    @EqualsAndHashCode.Include
    private Integer tourId;

    @Column(name = "tour_name", length = 150)
    private String tourName;

    @Column(name = "destination", length = 150)
    private String destination;

    @Column(name = "days")
    private Integer days;

    @Column(name = "nights")
    private Integer nights;

    @Column(name = "description", length = 1000)
    private String description;

    @Column(name = "price")
    private Float price;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    @ToString.Exclude
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subcat_id")
    @ToString.Exclude
    private SubCategoryMaster subCategory;

    @Column(name = "location", length = 150)
    private String location;

    @Column(name = "stay_and_meals", columnDefinition = "TEXT")
    private String stayAndMeals;

    @Column(name = "add_ons", columnDefinition = "TEXT")
    private String addOns;

    @Column(name = "passport_and_visa", columnDefinition = "TEXT")
    private String passportAndVisa;

    @Column(name = "weather", columnDefinition = "TEXT")
    private String weather;

    @Column(name = "do_and_dont", columnDefinition = "TEXT")
    private String doAndDont;

    @OneToMany(mappedBy = "tour", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @Builder.Default
    @ToString.Exclude
    private List<Schedule> schedules = new ArrayList<>();

    @OneToMany(mappedBy = "tour", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @Builder.Default
    @ToString.Exclude
    private List<Itinerary> itineraries = new ArrayList<>();

    @OneToMany(mappedBy = "tour", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @Builder.Default
    @ToString.Exclude
    private List<Journey> journeys = new ArrayList<>();

    @OneToMany(mappedBy = "tour", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @Builder.Default
    @ToString.Exclude
    private List<TourImages> images = new ArrayList<>();

    @OneToMany(mappedBy = "tour", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @Builder.Default
    @ToString.Exclude
    private List<Cost> costs = new ArrayList<>();

    @OneToMany(mappedBy = "tour", fetch = FetchType.LAZY)
    @Builder.Default
    @ToString.Exclude
    private List<Review> reviews = new ArrayList<>();
}

File: ETour-DotNet/Models/Tour.cs
namespace ETour.Api.Models;

public class Tour
{
    public int TourId { get; set; }
    public string TourName { get; set; }
    public string Destination { get; set; }
    public int? Days { get; set; }
    public int? Nights { get; set; }
    public string Description { get; set; }
    public float? Price { get; set; }

    public int? CategoryId { get; set; }
    public Category Category { get; set; }

    public int? SubcatId { get; set; }
    public SubCategoryMaster SubCategory { get; set; }

    public string Location { get; set; }
    public string TourType { get; set; }

    public string StayAndMeals { get; set; }
    public string AddOns { get; set; }
    public string PassportAndVisa { get; set; }
    public string Weather { get; set; }
    public string DoAndDont { get; set; }

    public ICollection<Schedule> Schedules { get; set; } = new List<Schedule>();
    public ICollection<Itinerary> Itineraries { get; set; } = new List<Itinerary>();
    public ICollection<Journey> Journeys { get; set; } = new List<Journey>();
    public ICollection<TourImages> Images { get; set; } = new List<TourImages>();
    public ICollection<Cost> Costs { get; set; } = new List<Cost>();
    public ICollection<TourCity> Cities { get; set; } = new List<TourCity>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}
